File: backend/src/domain/entities/employee.go
package entities

import "time"

type CreateEmployeeRequest struct {
	ShopID    string `json:"shop_id" bson:"shop_id"`
	FirstName string `json:"first_name" bson:"first_name" validate:"required"`
	LastName  string `json:"last_name" bson:"last_name" validate:"required"`
	Username  string `json:"username" bson:"username" validate:"required"`
	Password  string `json:"password" bson:"password" validate:"required,min=6"`
}

type UpdateEmployeeRequest struct {
	FirstName string `json:"first_name,omitempty" bson:"first_name,omitempty"`
	LastName  string `json:"last_name,omitempty" bson:"last_name,omitempty"`
	Username  string `json:"username,omitempty" bson:"username,omitempty"`
	Password  string `json:"password,omitempty" bson:"password,omitempty"`
}

type EmployeeResponse struct {
	EmployeeID string    `json:"employee_id"`
	ShopID     string    `json:"shop_id"`
	FirstName  string    `json:"first_name"`
	LastName   string    `json:"last_name"`
	Username   string    `json:"username"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type EmployeeListResponse struct {
	Employees  []EmployeeResponse `json:"employees"`
	TotalPages int                `json:"total_pages"`
	TotalItems int                `json:"total_items"`
	Page       int                `json:"page"`
	PageSize   int                `json:"page_size"`
}

File: frontend/src/types/employee.ts
export interface Employee {
  employee_id: string
  shop_id: string
  first_name: string
  last_name: string
  username: string
  created_at: string
  updated_at: string
}

export interface CreateEmployeeRequest {
  shop_id: string
  first_name: string
  last_name: string
  username: string
  password: string
}

export interface UpdateEmployeeRequest {
  first_name?: string
  last_name?: string
  username?: string
  password?: string
}

export interface EmployeeListResponse {
  message: string
  data: Employee[]
  page: number
  limit: number
  total_pages: number
  total_items: number
  status: number
}

export interface EmployeeResponse {
  message: string
  data: Employee
  status: number
}
